crates/ark/src/lsp/inputs/library.rs

```diff
@@ -0,0 +1,148 @@
+//
+// library.rs
+//
+// Copyright (C) 2025 by Posit Software, PBC
+//
+
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::sync::RwLock;
+
+use super::package::Package;
+use crate::lsp;
+
+/// Lazily manages a list of known R packages by name
+#[derive(Default, Clone, Debug)]
+pub struct Library {
+    /// Paths to library directories, i.e. what `base::libPaths()` returns.
+    pub library_paths: Arc<Vec<PathBuf>>,
+
+    packages: Arc<RwLock<HashMap<String, Option<Arc<Package>>>>>,
```

Comment:
```suggestion
    packages: Arc<RwLock<HashMap<String, Option<Arc<Package>>>>>,
```

Why is `Package` itself `Arc`ed?

I was assuming that `get()` would return an immutable reference to a `Package` in the package list

i.e. what `self.packages.read().unwrap().get(name)` already returns

And if it's not cached yet, you write it to the cache but still return a `get()` reference

Then it feels like you don't need to `Arc` the `Package` because `packages` would be the sole owner